src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
What ensures that the object outlives the sync code that makes use of it? Do we ensure that that `SyncConfig` outlives the `Realm`'s `SharedGroup`? Is the `SyncConfig` the right thing to be keeping this alive?

If it's up to some external entity to keep the transformer alive, why does `SyncConfig` share ownership? How does that external entity know when the transformer is no longer in use?